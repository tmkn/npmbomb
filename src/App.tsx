/** @jsx jsx */
import { jsx, css, Global, SerializedStyles } from "@emotion/react";
import React, { useState, lazy, Suspense } from "react";
import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";

const Index = lazy(() => import("./components/index/Index"));
const Package = lazy(() => import("./components/package/Package"));
const Results = lazy(() => import("./components/results/Results"));
const Overview = lazy(() => import("./components/overview/Overview"));

import { AppContext, IAppContext, IAppState, IGuessResult } from "./AppContext";
import Header from "./components/shared/header/Header";
import Content from "./components/shared/content/Content";
import Footer from "./components/shared/footer/Footer";
import { LoadingIndicator } from "./components/shared/loading/LoadingIndicator";
import { serifFont, sansSerifFont, primaryColor, secondaryColor, globalFocusStyle } from "./css";
import IconFont from "./codicon.ttf";
import "./codicon.css";

const globalStyle: SerializedStyles = css`
    @import url("https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@300&display=swap");

    body {
        background: white;
        margin: 0;
        padding: 0;
        min-height: 100vh;
        max-width: 100vw;
        display: flex;
        flex-direction: column;
        font-family: "${sansSerifFont}";
    }

    a {
        text-decoration: none;
    }

    h1 {
        font-family: "${serifFont}";
        color: ${primaryColor};
        font-weight: 200;
    }

    h2 {
        font-family: "${serifFont}";
        color: ${secondaryColor};
        font-weight: 200;
    }

    @font-face {
        font-family: "IconFont";
        src: url(${IconFont}) format("ttf");
    }

    @font-face {
        font-family: "Muli";
        font-style: normal;
        font-weight: 700;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/muli/v17/7Aulp_0qiz-aVz7u3PJLcUMYOFkpl0k30e6fwniDtzM.woff)
            format("woff");
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC,
            U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
    }
    @font-face {
        font-family: "Roboto Slab";
        font-style: normal;
        font-weight: 200;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/robotoslab/v10/BngbUXZYTXPIvIBgJJSb6s3BzlRRfKOFbvjoDISmb2RlV9Su1cai.woff)
            format("woff");
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC,
            U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
    }
    @font-face {
        font-family: "Roboto Slab";
        font-style: normal;
        font-weight: 600;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/robotoslab/v10/BngMUXZYTXPIvIBgJJSb6ufN5qWr4xCC.woff2)
            format("woff2");
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC,
            U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
    }

    @font-face {
        font-family: "Open Sans";
        font-style: normal;
        font-weight: 400;
        font-display: swap;
        src: local("Open Sans Regular"), local("OpenSans-Regular"),
            url(https://fonts.gstatic.com/s/opensans/v17/mem8YaGs126MiZpBA-UFVZ0bf8pkAg.woff2)
                format("woff2");
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC,
            U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
    }
`;

export const App: React.FC = () => {
    const [appState, setAppState] = useState<IAppState>({
        inGameMode: false,
        guesses: [],
        remaining: [],
        packages: []
    });

    const context: IAppContext = {
        appState,
        setAppState
    };

    return (
        <AppContext.Provider value={context}>
            <BrowserRouter>
                <Global styles={[globalStyle, globalFocusStyle]} />
                <Header />
                <Content>
                    <Switch>
                        <Route path="/package">
                            <Suspense fallback={<LoadingIndicator />}>
                                <Package />
                            </Suspense>
                        </Route>
                        <Route path="/results">
                            <Suspense fallback={<LoadingIndicator />}>
                                <Results />
                            </Suspense>
                        </Route>
                        <Route path="/overview">
                            <Suspense fallback={<LoadingIndicator />}>
                                <Overview />
                            </Suspense>
                        </Route>
                        <Route path="/">
                            <Suspense fallback={<LoadingIndicator />}>
                                <Index />
                            </Suspense>
                        </Route>
                        <Redirect to="/" />
                    </Switch>
                </Content>
                <Footer />
            </BrowserRouter>
        </AppContext.Provider>
    );
};
