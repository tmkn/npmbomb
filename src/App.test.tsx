import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import { App } from "./App";

describe("<ErrorComponent />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(<App />);

        expect(asFragment()).toMatchSnapshot();
    });
});
