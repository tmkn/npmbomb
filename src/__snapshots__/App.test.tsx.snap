// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorComponent /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-fmf8w1-Header"
  >
    <a
      href="/"
    >
      <div
        class="css-rcpacg-Header"
      >
        npmbomb
      </div>
    </a>
    <div
      class="css-vmd3lt-Header"
    >
      Guess NPM dependencies
    </div>
  </div>
  <div
    class="css-36v4m2-Content"
  >
    <div
      class="css-13ji2h0"
    >
      <h1>
        About
      </h1>
      <p
        class="css-1axajee"
      >
        Test your knowledge of the NPM ecosystem by guessing the number of dependencies for popular NPM packages.
      </p>
      <div
        class="css-8a5j6a"
      >
        <div
          class="css-16wgzhr"
        >
          Start
        </div>
      </div>
      <div
        class="css-14vnwjz"
        style="margin: 2rem 0px;"
      />
      <h2
        class="css-iio3hc"
      >
        FAQ
      </h2>
      <h2
        class="css-iio3hc"
      >
        <span
          style="cursor: pointer;"
        >
          <span
            class="css-swfxel-Faq"
          >
            &gt;
          </span>
           How is the dependency count calculated?
        </span>
      </h2>
      <div
        class="css-1rqbofr-Faq"
        style="display: none;"
      >
        As the dependency tree is traversed, each dependency from the dependencies field in the package.json is added to the count. It's a straight accumulation, as such a single package is counted multiple times if it is a dependency of other packages.
      </div>
      <h2
        class="css-iio3hc"
      >
        <span
          style="cursor: pointer;"
        >
          <span
            class="css-swfxel-Faq"
          >
            &gt;
          </span>
           Why is it called npmbomb?
        </span>
      </h2>
      <div
        class="css-1rqbofr-Faq"
        style="display: none;"
      >
        It's a hommage to zip bomb. According to Wikipedia a zip bomb is:
        <blockquote
          class="css-12z6zy1"
        >
          "A zip bomb is usually a small file .... however, when the file is unpacked, its contents are more than the system can handle."
        </blockquote>
        That sure sounds like your typical 
        <span
          class="css-18yn0ml-Highlight"
        >
          npm install
        </span>
         üòà
      </div>
      <h2
        class="css-iio3hc"
      >
        <span
          style="cursor: pointer;"
        >
          <span
            class="css-swfxel-Faq"
          >
            &gt;
          </span>
           Does it count dev dependencies?
        </span>
      </h2>
      <div
        class="css-1rqbofr-Faq"
        style="display: none;"
      >
        No, it only shows (and counts) dependencies from the 
        <span
          class="css-18yn0ml-Highlight"
        >
          dependencies
        </span>
         entry in the 
        <span
          class="css-18yn0ml-Highlight"
        >
          package.json
        </span>
         all the way to the very bottom.
      </div>
      <h2
        class="css-iio3hc"
      >
        <span
          style="cursor: pointer;"
        >
          <span
            class="css-swfxel-Faq"
          >
            &gt;
          </span>
           Why can't I find library XYZ?
        </span>
      </h2>
      <div
        class="css-1rqbofr-Faq"
        style="display: none;"
      >
        For now, it only shows a preselected number of libraries, namely the ones that are in the dependency tree of the most downloaded packages or specifically handpicked ones. If you want a specific dependency you can hit me up on Twitter 
        <a
          class="css-1qgk5q"
          href="https://twitter.com/tmkndev"
        >
          @tmkndev
        </a>
      </div>
      <h2
        class="css-iio3hc"
      >
        <span
          style="cursor: pointer;"
        >
          <span
            class="css-swfxel-Faq"
          >
            &gt;
          </span>
           Is it Open Source?
        </span>
      </h2>
      <div
        class="css-1rqbofr-Faq"
        style="display: none;"
      >
        <a
          class="css-1qgk5q"
          href="https://github.com/tmkn/npmbomb"
        >
          Yes
        </a>
      </div>
    </div>
  </div>
  <div
    class="css-kg3jfn-Footer"
  >
    <span>
      made with ‚ô•Ô∏è by 
      <a
        href="https://twitter.com/tmkndev"
      >
        @tmkndev
      </a>
    </span>
  </div>
</DocumentFragment>
`;
